Objective-C/Tests/ReplicatorTest+PendingDocIds.m

```diff
@@ -236,6 +237,49 @@ - (void) testPendingDocIdsWithFilter {
     [replicator removeChangeListenerWithToken: token];
 }
 
+- (void) testPendingDocIdsWhenOffline {
+    XCTestExpectation* x = [self expectationWithDescription: @"Replicator Stopped"];
+    CBLMutableDocument* doc = [self createDocument: @"doc-1"];
+    [doc setString: kCreateActionValue forKey: kActionKey];
+    [self saveDocument: doc];
+    
+    id target = [[CBLDatabaseEndpoint alloc] initWithDatabase: otherDB];
+    CBLReplicatorConfiguration* config = [self configWithTarget: target
+                                                           type: kCBLReplicatorTypePushAndPull
+                                                     continuous: YES];
+    
+    CBLReplicator* replicator = [[CBLReplicator alloc] initWithConfig: config];
+    __block CBLReplicator* r;
+    id<CBLListenerToken> token = [replicator addChangeListener: ^(CBLReplicatorChange* change) {
+        r = replicator;
+        NSError* err = nil;
+        NSSet* ids = [change.replicator pendingDocumentIDs: &err];
+        AssertNil(err);
+
+        if (change.status.activity == kCBLReplicatorConnecting) {
```

Comment:
I think that this test will fail if the states are updated in the wrong order.  That seems like a good thing.  I wonder, though, if you want to make that explicit: check to see that you first see CONNECTING, then IDLE, then OFFLINE.  Not necessary, I think, but it might make the test easier to read...